custom/ilsgateway/api.py

```diff
@@ -325,6 +332,69 @@ def web_user_sync(self, ilsgateway_webuser):
         web_user.save()
         return web_user
 
+    def _reassign_number(self, user, connection):
+        v = VerifiedNumber.by_phone(apply_leniency(connection.phone_number), include_pending=True)
+        if v.domain in self._get_logistics_domains():
+            v.domain = self.domain
+            v.owner_doc_type = user.doc_type
+            v.owner_id = user.get_id
+            backend_id = None
+            if connection.backend != 'push_backend':
+                backend_id = 'MOBILE_BACKEND_TEST'
+            v.backend_id = backend_id
+            v.verified = True
+            v.save()
+
+    def _save_verified_number(self, user, connection):
+        try:
+            self.save_verified_number(user, connection)
+        except PhoneNumberInUseException:
+            self._reassign_number(user, connection)
+        except InvalidFormatException:
+            pass
+
+    def save_verified_number(self, user, connection):
+        backend_id = None
+        if connection.backend != 'push_backend':
+            backend_id = 'MOBILE_BACKEND_TEST'
+        user.save_verified_number(self.domain, connection.phone_number, True, backend_id=backend_id)
+
+    def add_phone_numbers(self, ilsgateway_smsuser, user):
+        phone_numbers = ilsgateway_smsuser.phone_numbers
+        if not phone_numbers:
+            return
+
+        saved_numbers = []
+        for connection in phone_numbers:
+            phone_number = apply_leniency(connection.phone_number)
+            if connection.default:
+                saved_numbers = [phone_number] + saved_numbers
+            else:
+                saved_numbers.append(phone_number)
+        user.phone_numbers = saved_numbers
+        user.save()
+
+        for connection in phone_numbers:
+            self._save_verified_number(user, connection)
+
+    def edit_phone_numbers(self, ilsgateway_smsuser, user):
+        phone_numbers = set(user.phone_numbers)
+        saved_numbers = []
+
+        for connection in ilsgateway_smsuser.phone_numbers:
+            phone_number = apply_leniency(connection.phone_number)
+            if phone_number not in phone_numbers:
+                self._save_verified_number(user, connection)
+            if connection.default:
+                saved_numbers = [phone_number] + saved_numbers
+            else:
+                saved_numbers.append(phone_number)
+
+        for phone_number in phone_numbers - set(saved_numbers):
+            user.delete_verified_number(phone_number)
+
+        user.phone_numbers = saved_numbers
```

Comment:
I didn't see where this is called from, but do we need to save the user here like we do in `add_phone_numbers`?